hw/hw_07_task_03.py

```diff
@@ -0,0 +1,30 @@
+from typing import List, Union
+
+
+def check_win(board: List[List]) -> Union[int, str]:
+    win_coord = (
+        (0, 1, 2),
+        (3, 4, 5),
+        (6, 7, 8),
+        (0, 3, 6),
+        (1, 4, 7),
+        (2, 5, 8),
+        (0, 4, 8),
+        (2, 4, 6),
+    )
+    for each in win_coord:
+        combination = [board[i] for i in each]
+        if len(set(combination)) == 1 and combination[0] != "-":
+            return combination[0]
+    return 0
+
+
+def tic_tac_toe_checker(board: List[List]) -> str:
+    flatten_board = [val for sublist in board for val in sublist]
+    winner = check_win(flatten_board)
+    if isinstance(winner, str):
```

Comment:
It is a bad practice to bring type check to logic.  You can always use empty string as special value.